### PR TITLE
Sm/multiple-gitignore

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
- force isogit to be stricter about matching pkgDirs via a new `pathIsInFolder` function 
- look through the entire project for .gitignores and do more efficient stashing of them
- NUTs include some repro repos from FF

side effects
- STL shadowRepo is no longer a singleton (painful to test!)

testing:
test/nuts/repros has the customer provided projects that go with the repros on the gh issues.
yarn link this and use plugin-source to test.

### What issues does this PR fix or reference?
@W-10338491@ https://github.com/forcedotcom/cli/issues/1335 
@W-10338491@ https://github.com/forcedotcom/cli/issues/1336

### CODE REVIEW

File: src/shared/localShadowRepo.ts [hunk 1]
```diff
@@ -44,27 +45,25 @@ interface CommitRequest {
...
+  private gitIgnoreLocations: string[] = [];
 
   private constructor(options: ShadowRepoOptions) {
     this.gitDir = getGitDir(options.orgId, options.projectPath);
     this.projectPath = options.projectPath;
     this.packageDirs = options.packageDirs;
   }
 
+  // this is NOT a singleton but it was at one point, so its public method still sounds like it would be
   public static async getInstance(options: ShadowRepoOptions): Promise<ShadowRepo> {
```
Comment (WillieRuemmele) [inline src/shared/localShadowRepo.ts:63]: have you looked at current uses of this to make sure it isn't creating extra instances now that it's not a singelton?

Comment (mshanemc) [reply src/shared/localShadowRepo.ts:63]: It's not exported, so if someone besides `SourceTracking.ts` is using it, that's their own fault.

SourceTracking.ts checks that it only inits/loads a ShadowRepo once via `ensureLocalSourceTracking`.

But I have a better idea...hang on.

**Review by WillieRuemmele — APPROVED** [review]

**Comment by WillieRuemmele** [comment]
✅ 
1. create `force-app-two` directory
2. create classes within it
3. `beta:push` would push them, on this branch, it wouldn't

✅ : add a `.gitignore` file under `classes` and ignore a test class. `beta:push` and `beta:push -f` both deploy the "ignored" class. 
✅ : Tested with multiple `.gitignore` locations, where it is nested, classes in directories, how files were ignored

**Comment by mchadwickffdc** [comment]
@mshanemc great to seen this change has been taken into consideration and a fix has been merged into the source-tracking repo! 

I'm not familiar with the architecture or release process with the SFDX CLI, but when would expect to see this change go live in the production version of the SFDX CLI? 😊

**Comment by WillieRuemmele** [comment]
Hi @mchadwickffdc this is fixed in `7.133.0` the current `latest-rc` version, and will be promoted as `latest` next Thursday. Try installing that version to give it a whirl!
`npm install sfdx-cli@latest-rc -g`, or `sfdx update stable-rc`

**Comment by mchadwickffdc** [comment]
Thanks @WillieRuemmele, didn't know about these additional commands to get hold of the rc versions, i'll be sure to give it a whirl 😄

**Pull request merged**